enricher/api/src/main/java/io/fabric8/maven/enricher/api/BaseEnricher.java

```diff
@@ -105,6 +112,53 @@ protected EnricherContext getContext() {
         return enricherContext;
     }
 
+    protected void overrideEnvironmentVariables(KubernetesListBuilder builder, final Map<String, String> resourceEnv) {
+        builder.accept(new TypedVisitor<ContainerBuilder>() {
+            @Override
+            public void visit(ContainerBuilder containerBuilder) {
+                List<EnvVar> envVars = containerBuilder.buildEnv();
+
+                if (envVars == null) {
+                    envVars = new LinkedList<>();
+                    containerBuilder.withEnv(envVars);
+                }
+
+                for (Map.Entry<String, String> resourceEnvEntry : resourceEnv.entrySet()) {
+                    EnvVar newEnvVar =
+                        new EnvVarBuilder()
+                            .withName(resourceEnvEntry.getKey())
+                            .withValue(resourceEnvEntry.getValue())
+                            .build();
+                    if (hasEnvWithName(envVars, resourceEnvEntry.getKey())) {
+                        final String oldValue = removeEnvWithName(envVars, resourceEnvEntry.getKey());
+                        log.warn(
+                            "Environment variable %s is overridden: Setting the value %s, which replaces %s",
+                            newEnvVar.getName(), newEnvVar.getValue(), oldValue);
+                    }
+                    envVars.add(newEnvVar);
+                }
+            }
+        });
+
+    }
+
+    private String removeEnvWithName(List<EnvVar> envVars, String name) {
```

Comment:
Hey @lordofthejars , Can we make these two functions a generic one and include in into package io.fabric8.maven.core.util; ? This will make it reusable.  Maybe a new ListUtil.java class would be appropriate.

Reply:
Something like this?
removeEnvWithName --> removeItemWithName(List<T> itemList, String name)
hasEnvWithName --> hasItemWithName(List<T> itemList, String name)

Reply:
No, it cannot be generic because we need the `env.getName()` method which is not in a generic object. We could move this methods to another class, but not much more.

Reply:
Aah, I missed that!